internal/controller/promotions/promotions.go

```diff
@@ -665,3 +661,30 @@ func parseCreateActorAnnotation(promo *kargoapi.Promotion) string {
 	}
 	return creator
 }
+
+func calculateRequeueInterval(p *kargoapi.Promotion) time.Duration {
+	defaultRequeueInterval := 5 * time.Minute
+	for i := range p.Spec.Steps {
+		if p.Status.CurrentStep == int64(i) {
+			step := &promotion.Step{
+				Kind:   p.Spec.Steps[i].Uses,
+				Alias:  p.Spec.Steps[i].As,
+				If:     p.Spec.Steps[i].If,
+				Retry:  p.Spec.Steps[i].Retry,
+				Vars:   p.Spec.Steps[i].Vars,
+				Config: p.Spec.Steps[i].Config.Raw,
+			}
```

Comment:
Think much of this construction is unnecessary if my comment about `GetStepRunner` is taken into account.